### PR TITLE
developing html markup of site, added ErrorBoundary

### PR DESCRIPTION
**Pull request by lovelyserafima**

### CODE REVIEW

File: src/components/ErrorBoundary.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    static getDerivedStateFromError(error) {
+        // Update state so the next render will show the fallback UI.
+        return { hasError: true };
+    }
+
+    componentDidCatch(error, info) {
+        // Log error to the console
+        console.log(error, info);
```
Comment (yanaSelin) [inline src/components/ErrorBoundary.js]: Make sure, you log any debug info to console only in dev mode(take a glance at DefineWebpackPlugin for it)

File: src/components/constants/Constants.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+    DEFAULT_LIMIT: 12,
+    RELEASE_DATE: 'release date',
+    RATING: 'rating',
+
+    TITLE: 'title',
+    GENRES: 'genres',
+
+    MOVIES: 'movies',
+    DESC: 'desc'
```
Comment (yanaSelin) [inline src/components/constants/Constants.js]: It's better to export each member separately to don't prevent tree shaking

File: src/components/pages/searchpage/MainPage.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        const { sortingType, searchOption } = this.state;
+        fetchFromSearch(
+            searchString,
+            sortingTypeForSearch[sortingType],
+            searchOption
+        ).then(data => this.setState(state => ({ data: data.data })));
+    };
+
+    changeSearch = data => {
+        event.preventDefault();
```
Comment (yanaSelin) [inline src/components/pages/searchpage/MainPage.js]: To be honest I haven't seen this global variable before). And I think it's not really good option for 2 reasons:
1. it's deprecated
2. I think it's better to keep all logic related to form implementation in form component. There you have access to regular event object and can prevent default to encapsulate logic and don't make consumers care about it

File: src/util/sortingTypeConstants.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export const sortingTypeForSearch = {
```
Comment (yanaSelin) [inline src/util/sortingTypeConstants.js]: I think it's better to place all related helpers and constants to module folder(of course if they are not used somewhere else)

File: webpack.common.js [hunk 5]
```diff
@@ -13,14 +13,42 @@ module.exports = {
...
+                }
+            },
+            {
+                test: /\.(gif|png|jpe?g|svg)$/i,
+                use: [
+                    'file-loader',
+                    {
+                        loader: 'image-webpack-loader',
+                        options: {
+                            bypassOnDebug: true, // webpack@1.x
```
Comment (yanaSelin) [inline webpack.common.js]: as we use webpack4, it's redundant

File: src/components/ErrorBoundary.js (hunk 1 above)
Comment (lovelyserafima) [reply src/components/ErrorBoundary.js]: done

File: src/components/constants/Constants.js (hunk 2 above)
Comment (lovelyserafima) [reply src/components/constants/Constants.js]: done

File: src/components/pages/searchpage/MainPage.js (hunk 3 above)
Comment (lovelyserafima) [reply src/components/pages/searchpage/MainPage.js]: done

File: src/util/sortingTypeConstants.js (hunk 4 above)
Comment (lovelyserafima) [reply src/util/sortingTypeConstants.js]: done

File: webpack.common.js (hunk 5 above)
Comment (lovelyserafima) [reply webpack.common.js]: done

**Pull request merged**